### PR TITLE
Add metrics

### PR DESCRIPTION
**Pull request by black-adder**

### CODE REVIEW

**Comment by black-adder** [comment]
welllll, it looks like I missed a spot

**Comment by yurishkuro** [comment]
some easy wins for coverage is to exercise the noop impls. But the biggest unexercised chunk is reading from the timer channel. If you run it with high frequency, like 1ms, then you can also test it.

**Comment by yurishkuro** [comment]
btw, the actual build error is not about coverage (it does not block PRs), but from a test error

```
=== RUN   TestLocalMetrics
--- FAIL: TestLocalMetrics (0.00s)
	Error Trace:	local_test.go:15
			local_test.go:75
	Error:		Not equal: 3 (expected)
			        != 4 (actual)
	Messages:	Leaked at least one goroutine.
```
The class should have proper shutdown procedure when used in tests.

File: pkg/metrics/gauge.go [hunk 1]
```diff
@@ -0,0 +1,14 @@
+package metrics
```
Comment (yurishkuro) [inline pkg/metrics/gauge.go]: remove `pkg`, this whole repo is the equivalent of pkg namespace.

**Comment by yurishkuro** [comment]
Please read up on good commit mesages - http://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html

I updated the PR title

File: metrics/local.go [hunk 2]
```diff
@@ -0,0 +1,279 @@
...
+	tm       sync.RWMutex
+	counters map[string]*int64
+	gauges   map[string]*int64
+	timers   map[string]*localBackendTimer
+	stop     chan struct{}
+	stopped  chan struct{}
+}
+
+// NewLocalBackend returns a new LocalBackend
+func NewLocalBackend(tick time.Duration) *LocalBackend {
```
Comment (yurishkuro) [inline metrics/local.go]: nit: s/tick/collectionInterval/ and explain it in the doc string

File: metrics/local.go [hunk 3]
```diff
@@ -0,0 +1,279 @@
...
+	b := &LocalBackend{
+		counters: make(map[string]*int64),
+		gauges:   make(map[string]*int64),
+		timers:   make(map[string]*localBackendTimer),
+		stop:     make(chan struct{}),
+		stopped:  make(chan struct{}),
+	}
+
+	go func() {
```
Comment (yurishkuro) [inline metrics/local.go]: nit: move to a function

File: metrics/local.go [hunk 4]
```diff
@@ -0,0 +1,279 @@
...
+		*b.gauges[name] = value
+	} else {
+		atomic.StoreInt64(gauge, value)
+	}
+	b.gm.Unlock()
+}
+
+// RecordTimer records a timing duration
+func (b *LocalBackend) RecordTimer(name string, tags map[string]string, d time.Duration) {
+	timer := b.findOrCreateTimer(name)
```
Comment (yurishkuro) [inline metrics/local.go]: The tags are ignored here. That will make it impossible to use this local backend in unit tests that are sensitive to the tags. We can address this in a different PR, basically by copying functionality from the metrics implementation in jaeger-client-go.

**Pull request merged**